File: AtlasWeb/Core/Modules/Tibbi_Surec/Hemsirelik_Islemleri_Modulu/Hemsirelik_Is_Listesi/NursingWorkListViewModel.cs
using System;
using System.Linq;
using System.Net.Http;
using System.ComponentModel;
using TTInstanceManagement;
using Core.Models;
using TTObjectClasses;

using Infrastructure.Filters;
using System.Collections.Generic;
using Microsoft.AspNetCore.Mvc;


namespace Core.Models
{
    public partial class NursingWorkListViewModel
    {
        public OutputResource output
        {
            get;
            set;
        }

        public OutputDrugOrderDetail output_drugOrderDetails
        {
            get;
            set;
        }
        public List<PatientItem> patients
        {
            get;
            set;
        }
        public List<StatusItem> statusList
        {
            get;
            set;
        }

        public bool toolOption
        {
            get;
            set;
        }

        public OutputCaseOFNeedDrugOrder output_caseOfNeedDrugOrder
        {
            get;
            set;
        }

        public string ksMaterialNote { get; set; }
        public bool pharmcyOpen { get; set; }
    }
    public class OutputResource
    {
        public List<Resource> resources
        {
            get;
            set;
        }

        public DateTime startDate
        {
            get;
            set;
        }
        public DateTime endDate
        {
            get;
            set;
        }
    }

    public class OutputDrugOrderDetail
    {
        public List<DrugOrderDetailOutputItem> drugOrderDetails
        {
            get;
            set;
        }
    }


    public class OutputCaseOFNeedDrugOrder
    {
        public List<CostomDrugOrder> caseOfNeedDrugOrders
        {
            get;
            set;
        }
    }

    public class CostomDrugOrder
    {
        public string objectId { get; set; }
        public string PatientName { get; set; }
        public string DrugName { get; set; }

        public string DrugBarcode { get; set; }
        public string MasterResourceName { get; set; }

        public string DoseAmount { get; set; }

        public DateTime PlannedDateTime { get; set; }
    }






    public class InputFor_Get_DrugOrderDetails
    {
        public DateTime start_Time
        {
            get;
            set;
        }
        public DateTime end_Time
        {
            get;
            set;
        }

        public List<Guid> MasterResourcesList
        {
            get;
            set;
        }

        public int typeID { get; set; }
        public bool justMyPatient { get; set; }
    }

    public class DrugOrderDetailOutputItem
    {
        public Guid id
        {
            get;
            set;
        }
        public Guid stateDefID
        {
            get;
            set;
        }
        public string text
        {
            get;
            set;
        }

        public OrderTypeEnum typeId
        {
            get;
            set;
        }

        public string typeName { get; set; }

        public DateTime startDate
        {
            get;
            set;
        }

        public DateTime endDate
        {
            get;
            set;
        }

        public string statusName
        {
            get;
            set;
        }

        public DateTime periodStartTime
        {
            get;
            set;
        } //ana orderin başlama zamanı

        public bool? isChanged
        {
            get;
            set;
        } //takvim üzerinden değişiklik yapıldı mı

        public string doctorDescription
        {
            get;
            set;
        }

        public bool allDay
        {
            get;
            set;
        }

        public string Result
        {
            get;
            set;
        }

        public string StateID
        {
            get;
            set;
        }

        public string PatientFullName
        {
            get;
            set;
        }
        public string MasterResourceName
        {
            get;
            set;
        }
        public string BackColorName { get; set; }
    }

    public class InputFor_StateUpdateForSelecetedItem
    {
        public List<Guid> DrugOrderDetails
        {
            get;
            set;
        }
    }


    public class InputFor_UnnotifiedBaseTreatmentMaterialToUTS
    {
        public List<Guid> ResourceIds
        {
            get;
            set;
        }
    }

    public class PatientItem
    {
        public Guid ObjectID
        {
            get;
            set;
        }
        public string PatientFullName
        {
            get;
            set;
        }
        public string PatientUniqueRefNo
        {
            get;
            set;
        }
    }
    public class StatusItem
    {
        public Guid StateID
        {
            get;
            set;
        }
        public string StatusItemName
        {
            get;
            set;
        }

        public int TypeID { get; set; }
    }





}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Hemsirelik_Islemleri_Modulu/Hemsirelik_Is_Listesi/NursingWorkListViewModel.ts
//$DB89F690
import { BaseModel } from 'Fw/Models/BaseModel';
import { Guid } from 'NebulaClient/Mscorlib/Guid';
import { Type } from 'NebulaClient/ClassTransformer';
import { Resource } from 'NebulaClient/Model/AtlasClientModel';
import { OrderTypeEnum } from 'NebulaClient/Model/AtlasClientModel';
export class NursingWorkListViewModel extends BaseModel {
    @Type(() => OutputResource)
    output: OutputResource;
    @Type(() => OutputDrugOrderDetail)
    output_drugOrderDetails: OutputDrugOrderDetail;
    @Type(() => PatientItem)
    patients: Array<PatientItem>;
    @Type(() => StatusItem)
    statusList: Array<StatusItem>;
    toolOption: boolean;

    @Type(() => OutputCaseOFNeedDrugOrder)
    output_caseOfNeedDrugOrder: OutputCaseOFNeedDrugOrder;

    ksMaterialNote: string;
    pharmcyOpen: boolean;
}

export class OutputResource {
    public resources: Array<Resource>;
    @Type(() => Date)
    public startDate: Date;
    @Type(() => Date)
    public endDate: Date;
}

export class OutputDrugOrderDetail {
    @Type(() => DrugOrderDetailOutputItem)
    public drugOrderDetails: DrugOrderDetailOutputItem[];
}

export class OutputCaseOFNeedDrugOrder {
    @Type(() => CostomDrugOrder)
    public caseOfNeedDrugOrders: CostomDrugOrder[];
}

export class CostomDrugOrder {
    public objectId: string;
    public PatientName: string;
    public DrugName: string;
    public DrugBarcode: string;
    public MasterResourceName: string;
    public DoseAmount: string;
    @Type(() => Date)
    public PlannedDateTime: Date;
}


export class DrugOrderDetailOutputItem {
    @Type(() => Guid)
    public id: Guid;
    @Type(() => Guid)
    public stateDefID: Guid;
    public text: string;
    public typeId: OrderTypeEnum;
    public typeName: string;
    @Type(() => Date)
    public startDate: Date;
    @Type(() => Date)
    public endDate: Date;
    public statusName: string;
    @Type(() => Date)
    public periodStartTime: Date;
    @Type(() => Boolean)
    public isChanged: boolean;
    public doctorDescription: string;
    @Type(() => Boolean)
    public allDay: boolean;
    public Result: string;
    public StateID: string;
    public PatientFullName: string;
    public MasterResourceName: string;
    public BackColorName: string;
}

export class InputFor_Get_DrugOrderDetails {
    @Type(() => Date)
    public start_Time: Date;
    @Type(() => Date)
    public end_Time: Date;
    @Type(() => Guid)
    public MasterResourcesList: Guid[];
    @Type(() => Number)
    public typeID: number;
    @Type(() => Boolean)
    public justMyPatient: boolean;
}
export class InputFor_StateUpdateForSelecetedItem {
    @Type(() => Guid)
    public DrugOrderDetails: Array<Guid>;
}

export class PatientItem {
    @Type(() => Guid)
    ObjectID: Guid;
    PatientFullName: string;
    PatientUniqueRefNo: string;
}
export class StatusItem {
    @Type(() => Guid)
    StateID: Guid;
    StatusItemName: string;
    TypeID: number;

}
